diff --git a/libavcodec/x86/mlpdsp_init.c b/libavcodec/x86/mlpdsp_init.c
index 950f99683..c092752cc 100644
--- a/libavcodec/x86/mlpdsp_init.c
+++ b/libavcodec/x86/mlpdsp_init.c
@@ -45,7 +45,7 @@ void ff_mlp_rematrix_channel_##opt(int32_t *samples, \
 REMATRIX_CHANNEL_FUNC(sse4)
 REMATRIX_CHANNEL_FUNC(avx2_bmi2)
 
-#if HAVE_7REGS && HAVE_INLINE_ASM && HAVE_INLINE_ASM_NONLOCAL_LABELS
+#if HAVE_7REGS && HAVE_INLINE_ASM && HAVE_INLINE_ASM_NONLOCAL_LABELS && !defined(__clang__)
 
 extern char ff_mlp_firorder_8;
 extern char ff_mlp_firorder_7;
@@ -196,7 +196,7 @@ static void mlp_filter_channel_x86(int32_t *state, const int32_t *coeff,
 av_cold void ff_mlpdsp_init_x86(MLPDSPContext *c)
 {
     int cpu_flags = av_get_cpu_flags();
-#if HAVE_7REGS && HAVE_INLINE_ASM && HAVE_INLINE_ASM_NONLOCAL_LABELS
+#if HAVE_7REGS && HAVE_INLINE_ASM && HAVE_INLINE_ASM_NONLOCAL_LABELS && !defined(__clang__)
     if (INLINE_MMX(cpu_flags))
         c->mlp_filter_channel = mlp_filter_channel_x86;
 #endif
